worker/firewaller/firewaller.go

```diff
@@ -1100,61 +1126,238 @@ func diffRanges(currentRules, wantedRules []network.IngressRule) (toOpen, toClos
 	return toOpen, toClose
 }
 
+// relationLifeChanged manages the workers to process ingress changes for
+// the specified relation.
+func (fw *Firewaller) relationLifeChanged(tag names.RelationTag) error {
+	results, err := fw.remoteRelationsApi.Relations([]string{tag.Id()})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	relErr := results[0].Error
```

Comment:
Yeah, but sadly here and elsewhere the facade returns a slice of params results, each result struct containing a result and a params.Error. To fix it, we need to stop leaking params structs out of the facade layer.